src/app/give-help/material-aid-form/material-aid-form.component.ts

```diff
@@ -20,20 +24,38 @@ const CATEGORIES = Object.entries(MaterialAidOfferDefinitionDTO.CategoryEnum).ma
   styleUrls: ['./material-aid-form.component.scss'],
 })
 export class MaterialAidFormComponent {
-  data = defaults<MaterialAidOfferDefinitionDTO>({});
-  categories = CATEGORIES;
+  data = defaults<MaterialAidOfferDefinition>({});
+  CATEGORIES = CATEGORIES;
   PREFIXES = PREFIXES;
   phonePrefix: string = '48';
   phoneNumber: string = '';
-
-  constructor(private materialAidResourceService: MaterialAidResourceService) {}
+  OFFER_SENT_ALERT = OFFER_SENT_ALERT;
+  loading: boolean = false;
+  constructor(
+    private router: Router,
+    private materialAidResourceService: MaterialAidResourceService,
+    private snackbarService: SnackbarService
+  ) {}
 
   onPhoneNumberChange(): void {
-    // Waiting for TransportOfferDefinitionDTO receive a phoneNumber prop
-    // this.data.phoneNumber = this.phonePrefix + this.phoneNumber;
+    this.data.phoneNumber = this.phonePrefix + this.phoneNumber;
   }
 
   handleSubmit() {
-    this.materialAidResourceService.postMaterialAidOfferMaterialAid(this.data).subscribe((response) => {});
+    this.loading = true;
+    this.materialAidResourceService
+      .postMaterialAidOfferMaterialAid(this.data)
+      .pipe(take(1))
+      .subscribe(
```

Comment:
Maybe more readabke, concatmao for routingu, filtr operator for navigated, and only snackbar logic i subscribe?